packages/contracts/contracts/voting/token/TokenVoting.sol

```diff
@@ -70,27 +58,33 @@ contract TokenVoting is MajorityVotingBase {
         IDAO.Action[] calldata _actions,
         uint64 _startDate,
         uint64 _endDate,
-        bool _executeIfDecided,
-        VoteOption _choice
+        bool _tryEarlyExecution,
+        VoteOption _voteOption
     ) external override returns (uint256 proposalId) {
         uint64 snapshotBlock = getBlockNumber64() - 1;
 
         uint256 totalVotingPower = votingToken.getPastTotalSupply(snapshotBlock);
         if (totalVotingPower == 0) revert NoVotingPower();
 
-        if (votingToken.getPastVotes(_msgSender(), snapshotBlock) < minProposerVotingPower) {
+        if (votingToken.getPastVotes(_msgSender(), snapshotBlock) < minProposerVotingPower()) {
             revert ProposalCreationForbidden(_msgSender());
         }
 
         proposalId = proposalCount++;
 
         // Create the proposal
         Proposal storage proposal_ = proposals[proposalId];
-        (proposal_.startDate, proposal_.endDate) = _validateVoteDates(_startDate, _endDate);
-        proposal_.supportThreshold = supportThreshold;
-        proposal_.minParticipation = minParticipation;
-        proposal_.totalVotingPower = totalVotingPower;
-        proposal_.snapshotBlock = snapshotBlock;
+
+        (
+            proposal_.configuration.startDate,
+            proposal_.configuration.endDate
+        ) = _validateProposalDates(_startDate, _endDate);
+        proposal_.configuration.snapshotBlock = snapshotBlock;
+        proposal_.configuration.voteMode = voteMode();
```

Comment:
Do we need to store the vote mode for every proposal? 
- If the setting can't be globally changed then it's redundant
- If it can, then I see the need... but I think we're spending gas on storing data that will almost never change